distributed/shuffle/_memory.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from collections import defaultdict, deque
+from typing import Any, Callable
+
+from dask.sizeof import sizeof
+
+from distributed.shuffle._buffer import ShardsBuffer
+from distributed.shuffle._limiter import ResourceLimiter
+from distributed.utils import log_errors
+
+
+class MemoryShardsBuffer(ShardsBuffer):
```

Comment:
This can be refactored to avoid most of the baggage it inherits from `ShardsBuffer`. I'd do this in a follow-up since I expect us to rework buffers anyhow and don't want to spend time on API compatibility with the `DiskShardsBuffer` until then.